tools/syz-crush/crush.go

```diff
@@ -24,107 +28,252 @@ import (
 )
 
 var (
-	flagConfig = flag.String("config", "", "configuration file")
+	flagConfig      = flag.String("config", "", "configuration file")
+	flagRestartTime = flag.Duration("restartTime", 0, "restartPeriod how long to run the test.")
+	flagInfinite    = flag.Bool("infinite", true, "by default test is run for ever. -infinite=false to stop on crash")
 )
 
+type CrashReport struct {
+	vmIndex int
+	Report  *report.Report
+}
+
+type FileType int
+
+const (
+	LogFile FileType = iota
+	CProg
+)
+
+func getType(fileName string) FileType {
+	extension := filepath.Ext(fileName)
+
+	switch extension {
+	case ".c":
+		return CProg
+	case ".txt", ".log":
+		return LogFile
+	default:
+		log.Logf(0, "assuming logfile type")
```

Comment:
This can be simplified. THe only logic difference is only if the extension is ".c" or not. The log statement looks excessive, the decision about crepro/syzlog is inferrable from these lines below, which will appear in the log anyway:
```
			log.Logf(0, "compiled csource %v to cprog: %v", reproduceMe, cfg.SyzExecprogBin)
		log.Logf(0, "reproducing from logfile: %v", reproduceMe)
```